Mage/Archivist/VaultValue.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System;
+
+using Newtonsoft.Json.Linq;
+
+public class VaultValue {
+	private string _topic;
+	private JObject _index;
+	
+	private JToken _data;
+	private string _mediaType;
+	private uint _expirationTime;
+
+
+	//
+	public VaultValue(JObject valueObject) {
+		// Extract value data
+		_topic = valueObject["key"]["topic"].ToString();
+		_index = (JObject)valueObject["key"]["index"];
+
+		_data = JToken.Parse(valueObject["value"]["data"].ToString());
+		_mediaType = valueObject["value"]["mediaType"].ToString();
+		// TODO: handle expiration time
+		//_expirationTime = valueObject["value"]["expirationTime"];
+	}
+
+
+	// Property getters
+	public string topic { get { return _topic; } }
+	public JObject index { get { return _index; } }
+	
+	public JToken data { get { return _data; } }
+	public string mediaType { get { return _mediaType; } }
+	public uint expirationTime { get { return _expirationTime; } }
+
+
+	//
+	private JToken getProperty(JToken parent, JToken propertyName) {
+		return getProperty (parent, propertyName.ToString());
+	}
+
+	private JToken getProperty(JToken parent, string propertyName) {
+		if (parent.Type == JTokenType.Object) {
+			return parent[propertyName];
+		}
+
+		if (parent.Type == JTokenType.Array) {
+			return parent[int.Parse(propertyName)];
```

Comment:
I guess it's never been needed, and as this is a private API, don't worry about it.